mountpoint-s3-fs/src/fuse/config.rs

```diff
@@ -0,0 +1,223 @@
+use std::{
+    os::fd::RawFd,
+    path::{Path, PathBuf},
+};
+
+use anyhow::anyhow;
+use fuser::MountOption;
+use regex::Regex;
+
+/// Configuration for a FUSE background session.
+#[derive(Debug)]
+pub struct FuseSessionConfig {
+    pub(crate) mount_point: MountPoint,
+    pub(crate) options: Vec<MountOption>,
+    pub(crate) max_threads: usize,
+}
+
+impl FuseSessionConfig {
+    pub fn new(
+        mount_point: MountPoint,
+        read_only: bool,
+        auto_unmount: bool,
+        allow_root: bool,
+        allow_other: bool,
+        max_threads: usize,
+    ) -> anyhow::Result<Self> {
+        let fs_name = String::from("mountpoint-s3");
```

Comment:
Maybe we should move fs_name to a constant? I assume we might be using it other parts of the codebase as well, but we probably repeat it in multiple places